src/core/src/op/xor.cpp

```diff
@@ -2,105 +2,103 @@
+#include "shape_util.hpp"
 
-using namespace std;
-using namespace ngraph;
-
-op::v1::LogicalXor::LogicalXor(const Output<Node>& arg0,
-                               const Output<Node>& arg1,
-                               const AutoBroadcastSpec& auto_broadcast)
-    : BinaryElementwiseLogical(arg0, arg1, auto_broadcast) {
-    constructor_validate_and_infer_types();
-}
+namespace ov {
+namespace op {
+namespace logxor {
+struct Evaluate : element::NoAction<bool> {
+    using element::NoAction<bool>::visit;
 
-shared_ptr<Node> op::v1::LogicalXor::clone_with_new_inputs(const OutputVector& new_args) const {
-    OV_OP_SCOPE(v1_LogicalXor_clone_with_new_inputs);
-    check_new_args_count(this, new_args);
-    return make_shared<v1::LogicalXor>(new_args.at(0), new_args.at(1), this->get_autob());
-}
+    template <element::Type_t ET>
+    static result_type visit(const Tensor& arg0,
+                             const Tensor& arg1,
+                             Tensor& out,
+                             const AutoBroadcastSpec& broadcast_spec) {
+        using T = typename element_type_traits<ET>::value_type;
+        reference::logical_xor(arg0.data<const T>(),
+                               arg1.data<const T>(),
+                               out.data<T>(),
+                               arg0.get_shape(),
+                               arg1.get_shape(),
+                               broadcast_spec);
+        return true;
+    }
+};
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace logxor {
 namespace {
-template <element::Type_t ET>
-bool evaluate(const HostTensorPtr& arg0,
-              const HostTensorPtr& arg1,
-              const HostTensorPtr& out,
-              const op::AutoBroadcastSpec& broadcast_spec) {
-    ov::reference::logical_xor(arg0->get_data_ptr<ET>(),
-                               arg1->get_data_ptr<ET>(),
-                               out->get_data_ptr<ET>(),
-                               arg0->get_shape(),
-                               arg1->get_shape(),
-                               broadcast_spec);
-    return true;
+bool input_supported_type(const element::Type& et) {
+    return et == element::boolean;
 }
 
-bool evaluate_logxor(const HostTensorPtr& arg0,
-                     const HostTensorPtr& arg1,
-                     const HostTensorPtr& out,
-                     const op::AutoBroadcastSpec& broadcast_spec) {
-    bool rc = true;
-    out->set_broadcast(broadcast_spec, arg0, arg1);
-    switch (arg0->get_element_type()) {
-        NGRAPH_TYPE_CASE(evaluate_logxor, boolean, arg0, arg1, out, broadcast_spec);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
+bool evaluate(TensorVector& outputs, const TensorVector& inputs, const AutoBroadcastSpec& broadcast_spec) {
+    OPENVINO_ASSERT(outputs.size() == 1);
+    OPENVINO_ASSERT(inputs.size() == 2);
+
+    outputs[0].set_shape(ov::util::get_broadcast_shape(inputs[0].get_shape(), inputs[1].get_shape(), broadcast_spec));
```

Comment:
can we reuse shape infer?

Reply:
Will check.

Reply:
I think it can be replaced. 
In #19984 will add helper function which will work on ov::Shape and use shape inference to calculate shape. THen will replace get_broadcast_shape function.